src/DGEList.py

```diff
@@ -40,29 +39,43 @@ class DGEList(object):
     _field_strip_re = re.compile(r'[\s"]+')
 
     def __init__(
-        self,
-        counts: Optional[np.matrix] = None,
-        samples: Optional[np.array] = None,
-        genes: Optional[np.array] = None,
-        norm_factors: Optional[np.array] = None,
-        group: Optional[np.array] = None,
-        to_remove_zeroes: Optional[bool] = True
+            self,
+            counts: Optional[np.ndarray] = None,
+            samples: Optional[np.array] = None,
+            genes: Optional[np.array] = None,
+            norm_factors: Optional[np.array] = None,
+            group: Optional[np.array] = None,
+            to_remove_zeroes: Optional[bool] = False,
+            filename: Optional[str]=None,
     ) -> None:
-        if counts is None:
-            counts = np.matrix(np.zeros(3))
-        if norm_factors is None:
-            norm_factors = np.ones(np.size(counts, 1))
 
         self.to_remove_zeroes = to_remove_zeroes
-        self.counts = counts
-        self.samples = samples
-        self.genes = genes
-        self.norm_factors = norm_factors
-        self.group = group
+        if filename:
+            if counts or samples or genes or norm_factors or group:
+                raise Exception("if filename is provided, you can't also provide other parameters")
+            self._counts = None
+            self.import_file(filename)
```

Comment:
Make this a class method! Do not do this in `__init__()`. Follow the model under `read_handle()`.

Why do now have two conflicting methods for reading from a file-like object? Refactor into one.

Reply:
Not sure why you think they're conflicting.   Can you elaborate?